tpv_overwrite.yml

```
@@ -0,0 +1,840 @@
+
+  toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/search_gui/.*:
+    cores: 10
+    mem: 12
+    env:
+      _JAVA_OPTIONS: -XX:MaxPermSize=2G -Xmx6G -Xms1G -Djava.io.tmpdir=/data/2/galaxy_db/tmp -Duser.home=/data/2/galaxy_db/tmp
+    context:
+      name: special
+
+  toolshed.g2.bx.psu.edu/repos/earlhaminst/t_coffee/t_coffee/.*:
+    mem: 90
+    env:
+      DIR_4_TCOFFEE: $TMP
+      TMP_4_TCOFFEE: $TMP
+      CACHE_4_TCOFFEE: $TMP
+
+  toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/.*:
+    cores: 6
+    mem: 12
+    env:
+      _JAVA_OPTIONS: -XX:MaxPermSize=2G -Xmx6G -Xms1G -Djava.io.tmpdir=/data/2/galaxy_db/tmp -Duser.home=/data/2/galaxy_db/tmp
+    context:
+      name: special
+  toolshed.g2.bx.psu.edu/repos/iuc/trinity/trinity/.*:
+    cores: 4
+    mem: 512
+    env:
+      _JAVA_OPTIONS: -XX:MaxPermSize=2G -Xmx170G -Xms1G -Djava.io.tmpdir=/data/2/galaxy_db/tmp -Duser.home=/data/2/galaxy_db/tmp
+    context:
+      name: special
+
+  toolshed.g2.bx.psu.edu/repos/iuc/unicycler/unicycler/.*:
+    cores: 10
+    mem: 90
+    env:
+      _JAVA_OPTIONS: -XX:MaxPermSize=2G -Xmx6G -Xms1G -Djava.io.tmpdir=/data/2/galaxy_db/tmp -Duser.home=/data/2/galaxy_db/tmp
+      TERM: vt100
+    context:
+      name: special
+  toolshed.g2.bx.psu.edu/repos/imgteam/unzip/unzip/.*:
+    scheduling:
+      require:
+      - condor_singularity
+  .*xchem_pose_scoring.*:
+    cores: 1
+    gpus: 1
+    env:
+      CUDA_VISIBLE_DEVICES: 0
+    scheduling:
+      require:
+      - remote_condor_cluster_gpu_docker
+  .*run_jupyter_job.*:
+    cores: 8
+    mem: 32
+    scheduling:
+      require:
+      - condor_docker
+
+  .*stress_ng.*:
+    scheduling:
+      require:
+      - condor_singularity_with_conda
+
+  .*gatk2_base_recalibrator.*:
+    cores: 10
+    mem: 24
+    env:
+      _JAVA_OPTIONS: -XX:MaxPermSize=2G -Xmx8G -Xms1G -Djava.io.tmpdir=/data/2/galaxy_db/tmp
+    context:
+      name: _special
+
+  .*gatk2_reduce_reads.*:
+    cores: 10
+    mem: 24
+    env:
+      _JAVA_OPTIONS: -XX:MaxPermSize=2G -Xmx8G -Xms1G -Djava.io.tmpdir=/data/2/galaxy_db/tmp
+    context:
+      name: _special
+
+  .*echo_main_env.*:
+    context:
+      permissions:
+        allow:
+          roles:
+          - admin
+  .*_test_permissions_0.*: {}
+  .*_test_permissions_1.*:
+    rules:
+      - if: |
+          not user or user.email != "b@example.org"
+        fail: "The user does not have permission to run this tool"
+
+
+
+  .*_test_permissions_2.*:
+    rules:
+    
+    not user or 'Alphafold' not in [role.name for role in user.all_roles() if not role.deleted]
+    
+    
+      - if: |
+          not user or user.email != "a@example.org" or user not in roles.role-b
+        fail: "The user does not have permission to run this tool"
+
+
+
+
+
+
+    context:
+      permissions:
+        allow:
+          users:
+          - a@example.com
+          roles:
+          - role-b
```

Permissions will need to be implemented through tag matching as in this example: https://total-perspective-vortex.readthedocs.io/en/latest/topics/tpv_by_example.html#user-and-role-handling

```suggestion
roles:
  role-b:
    scheduling:
           accept:
             - authorize_dangerous_tool
```